cartoframes/columns.py

```diff
@@ -122,21 +125,35 @@ def normalize_name(column_name):
     return normalize_names([column_name])[0]
 
 
-def dtypes(columns, exclude_dates=False):
-    return {x.name: x.dtype for x in columns if not (exclude_dates is True and x.pgtype == 'date')}
+def dtypes(columns, exclude_dates=False, exclude_the_geom=False):
+    return {x.name: x.dtype if not x.name == 'cartodb_id' else 'int64'
+            for x in columns if not (exclude_dates is True and x.dtype in Column.DATETIME_DTYPES)
+            and not(exclude_the_geom is True and x.name in Column.SUPPORTED_GEOM_COL_NAMES)}
 
 
 def date_columns_names(columns):
-    return [x.name for x in columns if x.pgtype == 'date']
+    return [x.name for x in columns if x.dtype in Column.DATETIME_DTYPES]
 
 
 def pg2dtypes(pgtype):
     """Returns equivalent dtype for input `pgtype`."""
     mapping = {
-        'date': 'datetime64[ns]',
-        'number': 'float64',
-        'string': 'object',
+        'bigint': 'float64',
         'boolean': 'bool',
+        'date': 'datetime64[D]',
+        'double precision': 'float64',
         'geometry': 'object',
+        'int': 'int64',
+        'integer': 'float64',
+        'number': 'float64',
+        'numeric': 'float64',
+        'real': 'float64',
+        'smallint': 'float64',
+        'string': 'object',
+        'timestamp': 'datetime64[ns]',
+        'timestampz': 'datetime64[ns]',
+        'timestamp with time zone': 'datetime64[ns]',
+        'timestamp without time zone': 'datetime64[ns]',
+        'USER-DEFINED': 'object',
     }
     return mapping.get(str(pgtype), 'object')
```

Comment:
If we receive something unexpected, should not we let pandas decide what to use?

Reply:
Good point, as per how `dtypes` are processed in `read_csv` [it looks like we can just return `None`](https://github.com/pandas-dev/pandas/blob/master/pandas/io/parsers.py#L1657) and pandas will auto infer the data type.

Reply:
I made a quick test and it does not look it works nicely. For example is raising a ValueError when trying to autodetect a string (that looks like a JSON)

```
ValueError: could not convert string to float: '{"section/tags.au": "Australia", "source/au.geo.au-census": "Australian Bureau of Statistics (ABS)", "subsection/tags.boundary": "Boundaries", "license/au.geo.au-datapacks-license": "Creative Commons Attribution 2.5 Australia licence", "boundary_type/tags.cartographic_boundary": "Cartographic Boundary"}'
```

I'm reverting the change at the moment.

Reply:
:+1:

Reply:
We will be able to return to this and dates after some time if we detect bugs.